src/main/java/io/jenkins/plugins/okhttp/api/internals/JenkinsProxySelector.java

```diff
@@ -25,7 +25,7 @@ public class JenkinsProxySelector extends ProxySelector {
 
     public JenkinsProxySelector(final ProxyConfiguration configuration) {
         this.configuration = configuration;
-        this.proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(configuration.name, configuration.port));
+        this.proxy = new Proxy(Proxy.Type.HTTP, InetSocketAddress.createUnresolved(configuration.name, configuration.port));
```

Comment:
so if this was cached before then then we still have an issue.

as this is created per okHttpClient builder - if that is cached and re-used then whilst the resolution would happen when the proxy is used (🤞) it will not adapt to any changes in the Jenknis configuration.  (if you changed the jenkins confix from proxy1.acme.org to proxy2.acme.org you would still see access to proxy1

in other words the Proxy should really be obtained in the select - at which point it should not matter if you are using `createUnresolved` or not - and it will handle name changes (subject to java DNS cache - which you are still subject to here) as well as Proxy configuration changes.

Reply:
Correct. Which was my point in my second comment. If this is the preferred solution, I can update the PR.

Reply:
Sorry I completely missed that :)  

I would think it is preferred solution.   there will only be an extra DNS lookup if the DNS cache (java not the OS) has evicted a previous resolution - so if you call this rapidly then there should not be much overhead.  and if it is caused seldomly then there may be an extra DNS request - but this should not be called on any interactive path that matters,

Whilst the DNS resolution could be flaky due to bad infra and introduce some flakiness - if you have flaky DNS then you probably have bigger issues anyway...